hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/record/SingleTransaction.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.workflows.record;
+
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.node.base.Transaction;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A representation of a single transaction
+ */
+public interface SingleTransaction {
```

Comment:
So we can have both `SingleRecordTransaction` and `BlockStreamTransaction`. If both of these classes implement an interface, then we can just reference the interface in the handle workflow. 

Also, by using an interface instead of `SingleRecordTransaction`, we don't have to change every single place that uses `SingleRecordTransaction` yet. It's a way that a PR can be split so there aren't so many files changed, but without actually changing the functionality